### PR TITLE
Remove support for neo4j 4.2.*

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
This PR removes tests for Neo4j 4.2.* and updates documentation

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-1583.surge.sh

**Review by mjfwebb — DISMISSED** [review]
Looks good to me!

**Review by tbwiss — APPROVED** [review]

**Comment by darrellwarde** [comment]
Let's hold on off on merging this until tomorrow to allow any conversations or objections to elapse.

File: docs/modules/ROOT/pages/introduction.adoc [hunk 1]
```diff
@@ -67,12 +67,11 @@ Additionally, prerelease version numbers may have additional suffixes, for examp
 == Requirements
 
 1. https://neo4j.com/[Neo4j Database] with https://neo4j.com/developer/neo4j-apoc/[APOC] plugin:
```
Comment (darrellwarde) [inline docs/modules/ROOT/pages/introduction.adoc:69]: I think we can remove the subpoints and just trim this down to the following now!

```suggestion
1. https://neo4j.com/[Neo4j Database] 4.3.2+ with https://neo4j.com/developer/neo4j-apoc/[APOC] plugin:
```

**Review by darrellwarde — APPROVED** [review]
Just a small tweak to the docs, but I say let's get this merged!

File: docs/modules/ROOT/pages/introduction.adoc [hunk 2]
```diff
@@ -66,13 +66,12 @@ Additionally, prerelease version numbers may have additional suffixes, for examp
 [[introduction-requirements]]
 == Requirements
 
-1. https://neo4j.com/[Neo4j Database] with https://neo4j.com/developer/neo4j-apoc/[APOC] plugin:
-  - *4.2:* 4.2.9+ with APOC 4.2.0.8+
+1. https://neo4j.com/[Neo4j Database] 4.3.2+ with https://neo4j.com/developer/neo4j-apoc/[APOC] plugin:
   - *4.3:* 4.3.2+ with APOC 4.3.0.3+
   - *4.4:* 4.4.0+ with APOC 4.4.0.1+
```
Comment (darrellwarde) [inline docs/modules/ROOT/pages/introduction.adoc:71]: These can now be removed. 🙂 

```suggestion
```

**Review by darrellwarde — APPROVED** [review]
Just need to remove the trailing points!

**Review by darrellwarde — APPROVED** [review]
LGTM!

**Pull request merged**